crates/indexer/src/runner.rs

```diff
@@ -157,12 +162,28 @@ impl Runner {
 
     pub async fn run(&mut self) -> Result<()> {
         loop {
-            let result = self.insert().await?;
-
-            if !result {
-                let sleep_time = std::time::Duration::from_secs(3);
-                smol::Timer::after(sleep_time).await;
-            }
+            match self.insert().await {
+                Ok(result) => {
+                    if !result {
+                        let sleep_time = std::time::Duration::from_secs(1);
+                        smol::Timer::after(sleep_time).await;
+                    }
+                }
+                Err(err) => {
+                    let is_indexer_error = err.is::<IndexerError>();
+                    if is_indexer_error {
+                        let err_ref = err.downcast_ref::<IndexerError>();
```

Comment:
`err::is::<T>` is the same as `err::downcast_ref::<T>().is_some()`.

Reply:
updated